src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/caseload/ApproverCaseloadServiceTest.kt

```diff
@@ -41,359 +41,1030 @@ class ApproverCaseloadServiceTest {
 
-      verify(prisonApproverService, times(1)).getLicencesForApproval(aListOfPrisonCodes)
-      verify(probationSearchApiClient, times(1)).searchForPeopleByNomsNumber(nomisIds)
-      verify(prisonerSearchApiClient, times(1)).searchPrisonersByNomisIds(listOf("A1234AA", "B1234BB"))
-      verify(communityApiClient, times(1)).getStaffDetailsByUsername(comUsernames)
+        verify(prisonApproverService, times(1)).findRecentlyApprovedLicences(aListOfPrisonCodes)
+        verify(probationSearchApiClient, times(1)).searchForPeopleByNomsNumber(nomisIds)
+        verify(prisonerSearchApiClient, times(1)).searchPrisonersByNomisIds(nomisIds)
+        verify(communityApiClient, times(1)).getStaffDetailsByUsername(comUsernames)
 
-      assertThat(approvalCases).hasSize(2)
-      assertThat(approvalCases).extracting<Long> { it.licenceId }.containsExactly(2, 1)
-    }
-  }
+        assertThat(caseload).hasSize(2)
+        assertThat(caseload).extracting<Long> { it.licenceId }.containsExactly(3, 2)
+      }
 
-  @Test
-  fun `CADM prison caseload is filtered out`() {
-    val aListOfPrisonCodes = listOf("ABC", "DEF", "CADM")
-    service.getApprovalNeeded(aListOfPrisonCodes)
-    verify(prisonApproverService, times(1)).getLicencesForApproval(listOf("ABC", "DEF"))
-  }
+      @Test
+      fun `null release dates are surfaced above other release dates for the recently approved caseload`() {
+        val nomisIds = listOf("A1234AA", "B1234BB", "C1234CC")
+        val comUsernames = listOf("smills", "jdoe", "smills")
 
-  @Test
-  fun `Missing NOMIS ID on licence returns an empty caseload`() {
-    whenever(prisonApproverService.getLicencesForApproval(aListOfPrisonCodes)).thenReturn(
-      listOf(
-        aLicenceSummaryApproverView.copy(
-          nomisId = null,
-        ),
-      ),
-    )
+        whenever(prisonApproverService.findRecentlyApprovedLicences(aListOfPrisonCodes)).thenReturn(
+          listOf(
+            aLicenceSummaryApproverView.copy(
+              actualReleaseDate = LocalDate.of(2024, 6, 21),
+            ),
+            aLicenceSummaryApproverView.copy(
+              licenceId = 2L,
+              bookingId = 12345,
+              nomisId = "B1234BB",
+              forename = "John",
+              surname = "Doe",
+              prisonCode = "ABC",
+              prisonDescription = "ABC (HMP)",
+              comUsername = "jdoe",
+              licenceStatus = LicenceStatus.APPROVED,
+              actualReleaseDate = LocalDate.of(2024, 6, 20),
+            ),
+            aLicenceSummaryApproverView.copy(
+              licenceId = 3L,
+              bookingId = 67890,
+              nomisId = "C1234CC",
+              forename = "Jane",
+              surname = "Doe",
+              prisonCode = "MDI",
+              comUsername = "smills",
+              licenceStatus = LicenceStatus.ACTIVE,
+              actualReleaseDate = null,
+              conditionalReleaseDate = null,
```

Comment:
Should this licence summary be ordered first? 😅

Reply:
👍 cool sorry didn't grasp that at first

Reply:
Oh and didn't know you could just say in this order numerically instead of showing the response in the new order - nice!